ananse/network.py

```diff
@@ -368,54 +373,84 @@ def aggregate_binding(
                 "promoter region is larger than the maximum distance to use"
             )
 
-        # Get list of unique enhancers from the binding file
-        enhancer_pr = self.unique_enhancers(binding_fname)
+        hdf = HDFStore(binding_fname, "r")
 
-        # Link enhancers to genes on basis of distance to annotated TSS
-        gene_df = self.enhancer2gene(
-            enhancer_pr,
-            up=up,
-            down=down,
-            alpha=alpha,
-            promoter=promoter,
-            full_weight_region=full_weight_region,
-        )
-
-        # print(gene_df)
-        logger.info("Reading binding file...")
-        ddf = dd.read_csv(
-            binding_fname,
-            sep="\t",
-            usecols=["factor", "enhancer", "binding"],
-        )
-        if tfs is not None:
-            ddf = ddf[ddf["factor"].isin(tfs)]
-
-        # Merge binding information with gene information.
-        # This may be faster than creating index on enhancer first, but need to check!
-        tmp = ddf.merge(gene_df, left_on="enhancer", right_index=True)
-
-        # Remove everything with weight 0
-        tmp = tmp[tmp["weight"] > 0]
+        # TODO: This is hacky (depending on "_"), however the hdf.keys() method is
```

Comment:
if "\_" feels hacky, you could prefix the files with "ANANSE_". Either way, speed is king, so I'd agree.